org.adempiere.base/src/org/compiere/model/MUserDefTab.java

```diff
@@ -73,6 +80,14 @@ public static MUserDefTab getMatch (Properties ctx, int AD_Tab_ID, int AD_UserDe
 	{
 
 		MUserDefTab retValue = null;
+		
+		//  Check Cache
+		String key = new StringBuilder().append(AD_Tab_ID).append("_")
+				.append(AD_UserDefWin_ID)
+				.toString();
+		if (s_cache.containsKey(key))
+			return s_cache.get(key);
```

Comment:
you should either drop the ctx parameter or change it to s_cache.get(ctx, key, e->new MUserDefTab(ctx,e)) (will need to add copy constructor like what's being added to MUserDefWin). Without the copy constructor change, the method is inconsistent with regards to the ctx parameter - it might return PO with different ctx when it is retrieve from cache. Altenatively, removing the ctx parameter, you don't have to deal with this issue at all.